plugins/grid-bookmark/src/GridBookmarkWidget/components/DeleteBookmarksDialog.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import { observer } from 'mobx-react'
+import {
+  Button,
+  DialogContent,
+  DialogActions,
+  Alert,
+  List,
+  ListItemText,
+  ListItem,
+} from '@mui/material'
+import { Dialog } from '@jbrowse/core/ui'
+import { assembleLocString } from '@jbrowse/core/util'
+
+// locals
+import { GridBookmarkModel } from '../model'
+
+const DeleteBookmarksDialog = observer(function ({
+  onClose,
+  model,
+}: {
+  onClose: () => void
+  model: GridBookmarkModel
+}) {
+  const { selectedBookmarks } = model
+  const deleteAll = selectedBookmarks.length === 0
+
+  return (
+    <Dialog open onClose={onClose} title="Delete bookmarks">
+      <DialogContent>
+        <Alert severity="warning">
+          {deleteAll ? (
+            <>
+              <span>All bookmarks will be deleted.</span>
+              <br />
+              <span>
+                Use the checkboxes to select individual bookmarks to delete.
+              </span>
+            </>
+          ) : (
+            'Only selected bookmarks will be deleted.'
+          )}
```

Comment:
removed the list of bookmarks to be deleted, could be large, might be unnecessary to list in detail here